internal/cli/uninstall.go

```diff
@@ -55,24 +56,44 @@ func (c *UninstallCommand) Run(args []string) int {
 		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
 		return 1
 	}
-	c.ui.Output(
-		"Uninstalling Waypoint server with context %q",
-		contextDefault,
-		terminal.WithSuccessStyle(),
-	)
 
-	// Get the platform early so we can validate it.
-	p, ok := serverinstall.Platforms[strings.ToLower(c.platform)]
-	if !ok {
-		if c.platform == "" {
+	var ctxConfig *clicontext.Config
+	if contextDefault != "" {
+		ctxConfig, err = c.contextStorage.Load(contextDefault)
+		if err != nil {
+			c.ui.Output("Error loading context %q: %s", contextDefault, err.Error(), terminal.WithErrorStyle())
+			return 1
+		}
+	}
+
+	serverPlatform := c.platform
+	if ctxConfig.Server.Platform == "" && serverPlatform != "" {
```

Comment:
`ctxConfig` can be nil here according to if there is a default context or not, so we'll need to guard that.